pkg/controller/remote/cluster.go

```diff
@@ -25,23 +25,30 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// ClusterClientInterface is an interface encapsulating methods
+// to access a remote cluster.
+type ClusterClientInterface interface {
+	RESTConfig() *restclient.Config
+	CoreV1() (corev1.CoreV1Interface, error)
+}
+
 // ClusterClient is a helper struct to connect to remote workload clusters.
 type ClusterClient struct {
```

Comment:
Since you're adding an interface, how would you feel about renaming this to `clusterClient` and then having `ClusterClient` be the interface?

Reply:
Done